server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java

```diff
@@ -708,6 +710,39 @@ public void testNumericLongOrDateSortOptimization() throws Exception {
         dir.close();
     }
 
+    public void testIndexFieldHasDuplicateData() throws IOException {
```

Comment:
I wonder if we should use `BKDWriter/Reader` directly here since we could set the max number of docs per leave to a small value. This would ensure that we create enough leaves to test with fewer docs. Even with 10,000 docs here the number of leaves that we create is quite small (less than 10) so the test is a bit too simplistic in my opinion.

Reply:
@jimczi Thanks Jim for the review. This is addressed in the last commit